### ISSUE TITLE
Unlocking an Android screen when "Screen lock type" is None does not work

### ISSUE THREAD

**Issue by aaltat** [appium/appium#9462]
## Unlock screen does not work when "Screen lock type" is None

When using Appium 1.7.1 and starting Android settings, the Appium thinks that it is not able to unlock the screen. I have this code: 
```
from appium import webdriver

settings_caps = {"appPackage": "com.android.settings",
                 "appActivity": ".Settings",
                 "platformName": "android",
                 "platformVersion": "7.0",
                 "deviceName": "Galaxy J5 (2017)"}

driver = webdriver.Remote('http://localhost:4723/wd/hub', settings_caps)
driver.quit()
```
Then I see this in the logs: https://gist.github.com/aaltat/15db505eb62bc5e1096de264a8648156

## Environment
* Appium@1.7.1 and 1.7.0
* The problem does not occur in Appium@1.6.5:
* Running on Windows 7:
* Running NodeJs: v6.9.2
* Android 7.0
* Running reals device: Samsung J5 (2017)

## Details

When running the code above, I see that screen unlock but for some reason the Appium server thinks that it can not unlock the screen. 

## Link to Appium logs
Log from the Appium server are here: https://gist.github.com/aaltat/15db505eb62bc5e1096de264a8648156

**Comment by aaltat** [comment]
The problem seems the be related to Android OS version. The example code works by using Appium@1.7.1 and when using Android OS 6.0.1. But it does not work with using Appium@1.7.1 and Android OS 6.0.1

**Comment by mykola-mokhnach** [comment]
Check this tutorial: https://github.com/appium/appium-android-driver/blob/9a9ee19a71ba106d5afdfd3d98731881b7571b44/docs/UNLOCK.md

**Comment by aaltat** [comment]
I did, but I have two problems: 
- The tutorial does not explain how those parameters are actually used. Are they part of the capabilities or where/how one should use them?
- As far I understand the tutorial, the unlock type should be one of: 'pin', 'password', 'pattern', 'fingerprint'. My unlock type is not any of those.

**Comment by mykola-mokhnach** [comment]
@vrunoa Can you please advice?

**Comment by vrunoa** [comment]
@mykola-mokhnach already working on a solution. https://github.com/vrunoa/appium-android-driver/tree/add-swipe-to-unlock, We just need to swipe to unlock, already working on 5.1 and above, but something seems to be happing with 4.4 and touch actions. I'll try to have this fixed for the end of the week.

**Comment by aaltat** [comment]
My need is only for 6.0 or greater devices. 

Could the tutorial be enhanced to explain how those parameters are used, are they part of desired capabilities?

**Comment by vrunoa** [comment]
@aaltat i'll update the info here
https://github.com/appium/appium-android-driver/blob/master/docs/UNLOCK.md

**Comment by tfcardoso** [comment]
@aaltat I'm still waiting on vrunoa PR. In the meantime for those android devices before establishing the driver connection I run:

    adb -s «device_id» shell input keyevent 224

"Key code constant: Wakeup key. Wakes up the device. Behaves somewhat like KEYCODE_POWER but it has no effect if the device is already awake."

**Comment by aaltat** [comment]
Thanks for the update, I will definitely use that as a workaround.

**Comment by beingKhanvilkar** [comment]
I have tried using the below code to unlock the device:
 cap.setCapability("deviceName", "Redmi");
	  cap.setCapability("platformName", "Andorid");
	  cap.setCapability("platformVersion", "6.0.1");
	  cap.setCapability("unlockType", "pin");
	  cap.setCapability("unlockKey", "1111");
	  cap.setCapability("appPackage", "com.miui.player");
	  cap.setCapability("appActivity", "com.miui.player.ui.MusicBrowserActivity");

still I am getting the following error in appium logs:  Encountered internal error running command: Error: Error finding unlock pin buttons!

Please help to resolve this issue.

**Comment by OmarVector** [comment]
I've the same issue as beingKhanvilkar on the same android version.

**Comment by testPK** [comment]
I have the same issue as beingKhanvilkar , OmkarVektor and aaltar.
Tried it on Samsung Galaxy S9 and Android Version 8.

With or without pin I have the same issue.

**Comment by acanandraj** [comment]
@beingKhanvilkar 

Unlock is working. thanks dude.

**Comment by samuraii** [comment]
How can it be working if it is not working right now, same issue. 
Appium 1.6.2
Android 6, Huawei Honor

The pin screen appears, but then fails to find pin buttons

Encountered internal error running command: Error: Error finding unlock pin buttons!

**Comment by samuraii** [comment]
So the problem seems to be with latest available on appium webpage - 1.6.2, or maybe the device itself

**Comment by acanandraj** [comment]
Yes. That's why I have downgraded the appium version

**Comment by mykola-mokhnach** [comment]
There were some fixes done on this logic since Appium 1.10.0
